drivers/watchdog/wdt_mcux_wwdt.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2020, Jiri Kerestes
+ *
+ * Based on wdt_mcux_wdog32.c, which is:
+ * Copyright (c) 2019 Vestas Wind Systems A/S
+ * Copyright (c) 2018, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_lpc_wwdt
+
+#include <drivers/watchdog.h>
+#include <fsl_wwdt.h>
+#include <fsl_clock.h>
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wdt_mcux_wwdt);
+
+#define MIN_TIMEOUT 0xFF
+
+struct mcux_wwdt_config {
+	WWDT_Type *base;
+	uint8_t clk_divider;
+	void (*irq_config_func)(struct device *dev);
+};
+
+struct mcux_wwdt_data {
+	wdt_callback_t callback;
+	wwdt_config_t wwdt_config;
+	bool timeout_valid;
+};
+
+static int mcux_wwdt_setup(struct device *dev, uint8_t options)
+{
+	const struct mcux_wwdt_config *config = dev->config_info;
+	struct mcux_wwdt_data *data = dev->driver_data;
+	WWDT_Type *base = config->base;
+
+	if (!data->timeout_valid) {
+		LOG_ERR("No valid timeouts installed");
+		return -EINVAL;
+	}
+
+	WWDT_Init(base, &data->wwdt_config);
+	LOG_DBG("Setup the watchdog");
+
+	return 0;
+}
+
+static int mcux_wwdt_disable(struct device *dev)
+{
+	const struct mcux_wwdt_config *config = dev->config_info;
+	struct mcux_wwdt_data *data = dev->driver_data;
+	WWDT_Type *base = config->base;
+
+	WWDT_Deinit(base);
+	data->timeout_valid = false;
+	LOG_DBG("Disabled the watchdog");
+
+	return 0;
+}
+
+#define MSEC_TO_WWDT_TICKS(clock_freq, msec) \
+	((uint32_t)(clock_freq * msec / 1000U / 4))
```

Comment:
Please use the macro MSEC_PER_SEC ore something similar.
Also explain 4 with a comment, i.e. device divides source clock by 4.

Reply:
Done.